engine/classes/Elgg/I18n/Translator.php

```diff
@@ -188,18 +188,16 @@ public function addTranslation($country_code, $language_array) {
 		$country_code = strtolower($country_code);
 		$country_code = trim($country_code);
 
-		if (!is_array($language_array) || $country_code === "") {
+		if (!is_array($language_array) || empty($language_array) || $country_code === "") {
 			return false;
 		}
 
-		if (count($language_array) > 0) {
-			if (!isset($this->translations[$country_code])) {
-				$this->translations[$country_code] = $language_array;
-			} else {
-				$this->translations[$country_code] = $language_array + $this->translations[$country_code];
-			}
+		if (!isset($this->translations[$country_code])) {
+			$this->translations[$country_code] = [];
 		}
 
+		$this->translations[$country_code] = array_merge($this->translations[$country_code], $language_array);
```

Comment:
`array_merge` is a heavy computational function and it gets heavier the larger the array is. That's why a long loooong time ago this was changed to a `+` (see https://github.com/Elgg/Elgg/commit/7049bad2caedaad3aa84de9ba1141943bcec9772#diff-f67f4259ce99534800557ed80ab9015a)

Reply:
Very interesting. The difference in performance is considerable.
http://sandbox.onlinephpfunctions.com/code/78497c6594b2d8839064417e974117fbe792d7fb